plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/TrinoS3FileSystem.java

```diff
@@ -973,9 +979,40 @@ private AWSCredentialsProvider createAwsCredentialsProvider(URI uri, Configurati
                 .orElseGet(DefaultAWSCredentialsProviderChain::getInstance);
 
         if (iamRole != null) {
+            String stsEndpointOverwrite = conf.get(S3_STS_ENDPOINT);
```

Comment:
I'm not super familiar with the need for STS region/endpoint overrides. Is it only ever useful when an IAM role is used, or could it also be used without a role?

Reply:
Since STS service is used to assume a specific IAM role and get temporary credentials for that role I do not see any use case without setting an IAM role.